include/dali/core/tensor_shape.h

```diff
@@ -752,15 +793,13 @@ struct TensorListShape<DynamicDimensions>
   TensorListShape<other_ndim> to_static() const & {
     static_assert(other_ndim != DynamicDimensions,
                   "Conversion to static only allowed for static shape");
-    assert(sample_dim() == other_ndim && "Cannot convert to other ndim");
```

Comment:
What was wrong with that assert?

Reply:
It fails to convert an empty list.

Reply:
But it should be:
```c++
assert(num_samples() == 0 || sample_dim() == other_ndim && "Cannot convert to other ndim");
```

Reply:
But this would incorrectly pass, right?
```c++
TensorListShape<> tls;
tls.resize(0, 4);
tls.to_static<3>();
```

Reply:
I don't know how "incorrect" this is. It certainly can do no harm.